api/graphql/tests/test_reservation_units/test_reservation_units_query_opening_hours.py

```diff
@@ -0,0 +1,274 @@
+import datetime
+from zoneinfo import ZoneInfo
+
+import pytest
+from django.utils.timezone import get_default_timezone
+
+from reservation_units.models import ReservationUnit
+from tests.factories import (
+    OriginHaukiResourceFactory,
+    ReservableTimeSpanFactory,
+    ReservationUnitFactory,
+)
+
+# Applied to all tests
+pytestmark = [
+    pytest.mark.django_db,
+]
+
+DEFAULT_TIMEZONE = get_default_timezone()
+
+UTCOFFSET = f"0{int(datetime.datetime(2021, 5, 1, tzinfo=DEFAULT_TIMEZONE).utcoffset().total_seconds()/3600)}:00"
+
+
+@pytest.fixture()
+def reservation_unit() -> ReservationUnit:
+    origin_hauki_resource = OriginHaukiResourceFactory(
+        id="999",
+        opening_hours_hash="test_hash",
+        latest_fetched_date="2021-05-03",
+    )
+    return ReservationUnitFactory(origin_hauki_resource=origin_hauki_resource)
+
+
+def _get_date(*, day: int = 1, hour: int = 0, tzinfo: ZoneInfo | None = None):
+    return datetime.datetime(2023, 5, day, hour, 0, 0, tzinfo=tzinfo or DEFAULT_TIMEZONE)
+
+
+GET_QUERY = """
```

Comment:
How about using `tests.helpers.build_query` here? It would need some modification to enable filtering sub-selections, e.g. with `reservable_time_spans__start_date="2023-05-01"`.

Reply:
Adding filters to fields like so would require quite heavy modifications to the method, so I only changed these to use `build_query` by simply passing the filters in `fields` argument.

Reply:
OK, I can have a crack at implementing this elsewhere.